src/crypto.rs

```diff
@@ -545,8 +538,20 @@ impl PublicKey {
             })
         })?;
 
-        let key_id = calculate_key_id(der_bytes);
-        Ok(PublicKey { typ, key_id, scheme, value: PublicKeyValue(value) })
+        Self::new(typ, scheme, PublicKeyValue(value))
+    }
+
+    fn from_ed25519(bytes: Vec<u8>) -> Result<Self> {
+        if bytes.len() != 32 {
+            return Err(Error::IllegalArgument("ed25519 keys must be 44 bytes long".into()));
+        }
+
+        Ok(PublicKey {
```

Comment:
You use `PublicKey::new()` in `from_spki()`, you should probably do the same thing in both places.